pages/dao/[symbol]/proposal/components/instructions/Change/ChangeDonation.tsx

```diff
@@ -0,0 +1,282 @@
+          schema,
+          form,
+          programId,
+          connection,
+          wallet,
+          currentAccount: form.governedTokenAccount || null,
+          setFormErrors,
+        })
+      : getSolTransferInstruction({
+          schema,
+          form,
+          programId,
+          connection,
+          wallet,
+          currentAccount: form.governedTokenAccount || null,
+          setFormErrors,
+        })
+  }
+
+  useEffect(() => {
+    handleSetForm({
+      propertyName: 'programId',
+      value: programId?.toString(),
+    })
+  }, [realmInfo?.programId])
+  useEffect(() => {
+    if (form.destinationAccount) {
+      debounce.debounceFcn(async () => {
+        const pubKey = tryParseKey(form.destinationAccount)
+        if (pubKey) {
+          const account = await tryGetTokenAccount(connection.current, pubKey)
+          setDestinationAccount(account ? account : null)
+        } else {
+          setDestinationAccount(null)
+        }
+      })
+    } else {
+      setDestinationAccount(null)
+    }
+  }, [form.destinationAccount])
+  useEffect(() => {
+    handleSetInstructions(
+      { governedAccount: governedAccount, getInstruction },
+      index
+    )
+  }, [form])
+  useEffect(() => {
+    setGovernedAccount(form.governedTokenAccount?.governance)
+    setMintInfo(form.governedTokenAccount?.extensions.mint?.account)
+  }, [form.governedTokenAccount])
+  const destinationAccountName =
+    destinationAccount?.publicKey &&
+    getAccountName(destinationAccount?.account.address)
+  const schema = getTokenTransferSchema({ form, connection })
+
+  return (
+    <>
+      <GovernedAccountSelect
+        label="Source account"
+        //currently limiting to SOL deposits
+        governedAccounts={governedTokenAccountsWithoutNfts.filter(
+          (governedTokenAccount) => {
+            return governedTokenAccount.isSol
+          }
+        )}
+        onChange={(value) => {
+          handleSetForm({ value, propertyName: 'governedTokenAccount' })
+        }}
+        value={form.governedTokenAccount}
+        error={formErrors['governedTokenAccount']}
+        shouldBeGoverned={shouldBeGoverned}
+        governance={governance}
+      ></GovernedAccountSelect>
+      <NonprofitSelect
+        value={searchInput}
+        onSearch={handleSearch}
+        onSelect={(nonprofitName) => handleSelectNonProfit(nonprofitName)}
+        className="h-12"
+        showSearchResults={searchResults.length > 0 || isTyping}
+        disabled={searchResults.length === 0}
+        nonprofitInformation={selectedNonprofit}
+        isLoading={loading || isTyping}
+      >
+        {searchResults.map((foundNonprofit) => (
+          <NonprofitSelect.Option
+            key={foundNonprofit.ein}
+            value={foundNonprofit.name}
+          >
+            <span>{foundNonprofit.name}</span>
+          </NonprofitSelect.Option>
+        ))}
+      </NonprofitSelect>
+      {destinationAccount && (
+        <div>
```

Comment:
rather than conditionally create components, it's more efficient to create them always and then conditionally hide. Looks like you're using tailwind, so something like

```
<div className={`${!destinationAccount && "hidden"}`}>...</div>
```

on your wrapper `div` here